src/components/modebar/modebar.js

```diff
@@ -45,13 +45,28 @@ var proto = ModeBar.prototype;
 proto.update = function(graphInfo, buttons) {
     this.graphInfo = graphInfo;
 
-    var context = this.graphInfo._context;
+    var context = this.graphInfo._context,
+        fullLayout = this.graphInfo._fullLayout,
+        modeBarId = 'modebar-' + fullLayout._uid;
+
+    this.element.setAttribute('id', modeBarId);
+    this._uid = modeBarId;
 
     if(context.displayModeBar === 'hover') {
         this.element.className = 'modebar modebar--hover';
     }
     else this.element.className = 'modebar';
 
+    if(fullLayout.modebar.orientation === 'v') {
+        this.element.className += ' vertical';
+        buttons = buttons.reverse();
+    }
+
+    Lib.addRelatedStyleRule(modeBarId, '#' + modeBarId, 'background-color: ' + fullLayout.modebar.bgcolor);
+    Lib.addRelatedStyleRule(modeBarId, '#' + modeBarId + ' .modebar-btn .icon path', 'fill: ' + fullLayout.modebar.color);
+    Lib.addRelatedStyleRule(modeBarId, '#' + modeBarId + ' .modebar-btn:hover .icon path', 'fill: ' + fullLayout.modebar.activecolor);
+    Lib.addRelatedStyleRule(modeBarId, '#' + modeBarId + ' .modebar-btn.active .icon path', 'fill: ' + fullLayout.modebar.activecolor);
```

Comment:
These don't appear to update properly on `relayout` e.g. `Plotly.relayout(gd, 'modebar.bgcolor', 'blue')`

Reply:
![peek 2018-10-04 13-54](https://user-images.githubusercontent.com/6675409/46492957-0b889380-c7dd-11e8-869e-f46d12851756.gif)

moreover, `modebar.color` doesn't seem to work on some icons.

Reply:
Let me investigate `relayout`!

As for the screen capture: the gray icons are the ones that are currently active (gray being the default `activecolor`). It does look like a bug the first time you try it but it's not. Maybe I should implement what @nicolaskruchten suggested over here: https://github.com/plotly/plotly.js/issues/3063#issuecomment-426720053. This shouldn't be a problem in Dash however since they usually explicitly style everything.

Reply:
> the gray icons are the ones that are currently active (gray being the default `activecolor`).

Ha I see. My mistake!

Reply:
Commit 97f25e2 fixes the issue with `relayout`. 

The style rules were just appended to the same `style` element and would have less priority than previous rules. This is fixed by starting with a fresh `style` element each time.

Reply:
Cool. Let's :lock: this down in a jasmine test though.

Reply:
In commit f5cc0e9, we test each new layout attributes and make sure they behave properly when using `relayout`!

Reply:
Great tests. Thanks :tada: